app/src/main/java/com/github/se/polyfit/ui/viewModel/GraphViewModel.kt

```diff
@@ -25,29 +30,87 @@ enum class SortPoints {
   }
 }
 
+enum class DisplayScreen {
+  OVERVIEW,
+  GRAPH_SCREEN
+}
+
 enum class SortDirection {
   ASCENDING,
   DESCENDING
 }
 
 @HiltViewModel
-class GraphViewModel @Inject constructor() : ViewModel() {
+class GraphViewModel @Inject constructor(private val dataProcessor: LocalDataProcessor) :
+    ViewModel() {
   private val _sortedPoints = MutableLiveData(SortPoints.KCAL)
 
   private val _sortDirection = MutableLiveData(SortDirection.ASCENDING)
 
   private val _searchText = MutableLiveData("")
   val searchText: LiveData<String> = _searchText
 
-  private val _graphData = MutableLiveData(mockData)
-  val graphData = updateGraphData()
+  private val _graphData = MediatorLiveData<List<GraphData>>()
+  val graphData = _graphData
+
+  init {
+    fetchCaloriesData()
+  }
+
+  private fun fetchCaloriesData(numDays: Int = 7) {
+    viewModelScope.launch(Dispatchers.IO) {
+      val today = LocalDate.now()
+      // Create a list of dates for the last 7 days including today
+      val dates = (0 ..< numDays).map { today.minusDays(it.toLong()) }
+      // Create a baseline list of GraphData with calories set to 0
+      val baselineGraphData =
+          dates.map { GraphData(kCal = 0.0, date = it, weight = 0.0) }.toMutableList()
+
```

Comment:
this code is identical to initGraphData except that it uses `numDays`. Why not just make `initGraphData` follow the same format of having a parameter that is optional, and then call that function here, to avoid the code duplication?

e.g.

```
  fun initGraphData(numDays: Int = 7): MutableList<GraphData> {
       val today = LocalDate.now()
      // Create a list of dates for the last 7 days including today
      val dates = (0 ..< numDays).map { today.minusDays(it.toLong()) }
      // Create a baseline list of GraphData with calories set to 0
      val baselineGraphData =
          dates.map { GraphData(kCal = 0.0, date = it, weight = 0.0) }.toMutableList()

       return baselineGraphData
  }
```

and then here:

```
viewModelScope.launch(Dispatchers.IO) {
      val baselineGraphData = initGraphData(numDays)
      ....
```

so that you can avoid duplicate code